docs/rfcs/1011-non-blocking-invite-transport.md

```diff
@@ -0,0 +1,1231 @@
+- the SAS code communicated to the user was invalid
+- the payload could not be deciphered
+- the payload could not be deserialized
+- the payload contained invalid information
+- the attempt has been automatically cancelled by a new `start_attempt` command
+
+It is also registered that the greeter cancelled the attempt, so that this information can be provided in subsequent `invitation_cancelled` replies, along with the corresponding reason and timestamp. This way, the front-end application has all the information to properly communicate to the user what happened during the attempt.
+
+Pseudo code:
+
+```python
+invite_greeter_cancel_attempt(
+    token: InvitationToken,
+    attempt: AttemptID,
+    reason: CancelledAttemptReason,
+
+)
+->
+  InvitationNotFound | InvitationCompleted | InvitationCancelled
+  AttemptNotFound | AttemptNotJoined | AttemptCancelled
+  GreeterNotAllowed | OK:
+    # Perform checks
+    [...]
+    # Call `greeter_cancel_attempt(reason)`
+    [...]
+```
+
+Schema definition:
+
+```json5
+[
+    {
+        "major_versions": [
+            4
+        ],
+        "req": {
+            "cmd": "invite_greeter_cancel_attempt",
+            "fields": [
+                {
+                    "name": "token",
+                    "type": "InvitationToken"
+                },
+                {
+                    "name": "attempt",
+                    "type": "AttemptID"
+                },
+                {
+                    "name": "reason",
+                    "type": "CancelledAttemptReason"
+                }
+            ]
+        },
+        "reps": [
+            {
+                "status": "ok",
+                "fields": [
+                    {
+                        "name": "attempt",
+                        "type": "AttemptID"
+                    }
+                ]
+            },
+            {
+                // The invitation token doesn't correspond to any existing invitation
+                "status": "invitation_not_found"
+            },
+            {
+                // The invitation has already been completed
+                "status": "invitation_completed"
+            },
+            {
+                // The invitation has been cancelled
+                "status": "invitation_cancelled"
+            },
+            {
+                // The author is not part of the allowed greeters for this invitation
+                "status": "author_not_allowed"
+            },
+            {
+                // The attempt id doesn't correspond to any existing attempt
+                "status": "attempt_not_found"
+            },
+            {
+                // The author did not join the attempt
+                // This should not happen, since joining is required to get the attempt ID
+                "status": "attempt_not_joined"
+            },
+            {
+                // The attempt has been already cancelled
+                "status": "attempt_already_cancelled",
```

Comment:
`already` is there when needed to avoid confusion (e.g `invitiation_already_completed` is more clear than `invitation_completed` when calling `invite_complete`)